### PR TITLE
chef: Support Node.js zero config for Linux

### PR DESCRIPTION
**Pull request by jeffreyc-splunk**
- Add initial Node.js zero config for Linux support with chef (installed and activated by default if `npm` is found)
- Requires Node.js and `npm` to be installed/managed separately
- Update tests to install node with the `nodejs` cookbook, verify installation of `splunk-otel-js`, and verify config files for both preload and systemd methods

### CODE REVIEW

File: deployments/chef/kitchen.yml [hunk 1]
```diff
@@ -200,19 +211,169 @@ suites:
...
         with_auto_instrumentation: true
-        auto_instrumentation_ld_so_preload: "# this should not be added"
+        auto_instrumentation_systemd: true
+        auto_instrumentation_ld_so_preload: "# my extra library"
+        auto_instrumentation_resource_attributes: deployment.environment=test
+        auto_instrumentation_service_name: test
+        auto_instrumentation_enable_profiler: true
+        auto_instrumentation_enable_profiler_memory: true
+        auto_instrumentation_enable_metrics: true
+        auto_instrumentation_otlp_endpoint: http://0.0.0.0:4317
```
[the remark this answers is not in the document]
Comment (jeffreyc-splunk) [reply deployments/chef/kitchen.yml:230]: This file and the values defined are just for tests.

**Review by jvoravong — APPROVED** [review]

**Comment by jeffreyc-splunk** [comment]
> If the `0.0.0.0` is correct I think it's ready to ship. Could be cleaner but can always take care of that in a subsequent review, none of the nits block shipment (hence, `nit`'s and not `blocker`'s)

The kitchen.yml file and values defined within are only for tests.

File: deployments/chef/kitchen.yml [hunk 2]
```diff
@@ -200,19 +211,169 @@ suites:
...
+        *nodejs_options
+      splunk_otel_collector:
+        splunk_access_token: testing123
+        splunk_realm: test
+        with_auto_instrumentation: true
+        with_auto_instrumentation_sdks:
+          - nodejs
+        auto_instrumentation_systemd: true
+
+  - name: with_custom_systemd_node_instrumentation
```
[the remark this answers is not in the document]
Comment (jeffreyc-splunk) [reply deployments/chef/kitchen.yml:341]: It does (see https://github.com/signalfx/splunk-otel-collector/pull/4020/files#diff-6eaaa2b600b17329126f81fa34ce2ffbac5c26a5b58c78efba154cfb195d9543R169), but there are differences in each of the test scenarios.

File: deployments/chef/kitchen.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeffreyc-splunk) [reply deployments/chef/kitchen.yml:237]: The default is still `http://127.0.0.1:4317` (see https://github.com/signalfx/splunk-otel-collector/pull/4020/files#diff-daed86d2dd43f673c02b845b510649554fcaa9d78cdaa3835ed570c1cb5553a3R102).  This is just a custom value for tests.

**Pull request merged**